core/templates/components/summary-tile/story-summary-tile.component.html

```diff
@@ -1,119 +1,118 @@
 <div class="story-summary-tile protractor-test-story-summary-title">
   <div class="story-progress-bar">
-    <svg class="story-progress-svg" width="50" height="50" ng-if="$ctrl.nodeCount">
+    <svg class="story-progress-svg" width="50" height="50" *ngIf="nodeCount">
       <circle class="story-progress-circle total-circle"
               cx="25"
               cy="25"
               r="20"
-              stroke-dasharray="<[$ctrl.getStrokeDashArrayValues()]>"
+              [attr.stroke-dasharray]="getStrokeDashArrayValues()"
               stroke-dashoffset="-2.5"
               stroke-linecap="round"/>
       <circle class="story-progress-circle completed-circle"
               cx="25"
               cy="25"
               r="20"
-              stroke-dasharray="<[$ctrl.getCompletedStrokeDashArrayValues()]>"
+              [attr.stroke-dasharray]="getCompletedStrokeDashArrayValues()"
               stroke-dashoffset="-2.5"
-              stroke-linecap="<[$ctrl.completedStoriesCount ? 'round' : 'butt']>"/>
+              [attr.stroke-linecap]="completedStoriesCount ? 'round' : 'butt'"/>
       <text x="25"
             y="-20"
             text-anchor="middle"
             stroke="#000"
             stroke-width="0.5"
             font-size="0.9em"
             transform="rotate(90)">
-        <[$ctrl.storyProgress]>%
+        {{ storyProgress }}%
       </text>
     </svg>
   </div>
   <div class="oppia-story-title">
     <strong class="title-text">
-      <a ng-href="<[$ctrl.getStoryLink()]>" ng-class="{'disabled-link': !$ctrl.classroomUrlFragment}"><[$ctrl.storySummary.getTitle()]> </a>
+      <a [href]="getStoryLink()" [ngClass]="{'disabled-link': !classroomUrlFragment}">{{ storySummary.getTitle() }} </a>
     </strong>
     <span class="oppia-chapters-count">
-      <span translate="I18N_TOPIC_VIEWER_CHAPTERS"
-            translate-values="{count: $ctrl.nodeCount}"
-            translate-interpolation="messageformat">
+      <span>
+        {{ nodeCount }} chapters
       </span>
     </span>
   </div>
   <div class="story-details">
     <div class="story-thumbnail"
-         ng-style="{background: $ctrl.storySummary.getThumbnailBgColor()}">
-      <a ng-href="<[$ctrl.getStoryLink()]>">
-        <img ng-src="<[$ctrl.thumbnailUrl]>"
+         [ngStyle]="{background: storySummary.getThumbnailBgColor()}">
+      <a [href]="getStoryLink()">
+        <img [src]="thumbnailUrl"
              alt=""
              class="img-thumbnail"
-             ng-style="{background: $ctrl.storySummary.getThumbnailBgColor()}">
+             [ngStyle]="{background: storySummary.getThumbnailBgColor()}">
       </a>
     </div>
     <div class="chapter-details">
       <div class="chapter-title"
-           ng-repeat="title in $ctrl.storySummary.getNodeTitles()">
-        <div ng-if="$ctrl.storySummary.isNodeCompleted(title)" class="material-icons completed-icon">
+           *ngFor="let title of storySummary.getNodeTitles();let index = index">
```

Comment:
can't we use the index directly instead of redefining it? (not sure!)

Reply:
Either we have to do it like `index as index` or the above way but we have to define it.

Reply:
Maybe use `let idx = index` so that it is less confusing.